wagtail/admin/views/pages/edit.py

```diff
@@ -647,8 +656,10 @@ def submit_action(self):
             previous_revision=(self.previous_revision if self.is_reverting else None)
         )
 
-        self.log_commenting_changes(revision)
-        self.send_commenting_notifications()
+        if 'comments' in self.form.formsets:
```

Comment:
We save the page in a few other places, including `perform_workflow_action`, but also on workflow cancel/restart. We should make sure changes are logged and notifications sent there too (and probably consolidate some of this)